Merge release/19.3 (code freeze / 19.3-rc-1)

AliSoftware opened the pull request:
- [x] New empty entry created in `RELEASE-NOTES.txt` for next iteration.
- [x] `{jetpack_}metadata/release_notes.txt` extracted for WordPress and Jetpack.
- [x] `WordPress/jetpack_metadata/release_notes.txt` audited to remove any item not applicable for Jetpack. — see discussions below
- [x] Internal dependencies (FluxC, Login. Stories. About…) bumped to a new stable version in `build.gradle` if necessary.
- [x] ~`WordPress/src/main/res/values/strings.xml` updated with strings from binary dependencies.~ — N/A: No new strings introduced by those new versions
- [x] New strings frozen/copied into `fastlane/resources/values/strings.xml`.
- [x] Version bumped in `version.properties`

AliSoftware committed:
Bump version number

AliSoftware committed:
Clean up release files.

AliSoftware committed:
Update draft release notes for Jetpack 19.3.

AliSoftware committed:
Release Notes: add new section for next version (19.4)

AliSoftware committed:
Update draft release notes for 19.3.

AliSoftware committed:
Bump libs to stable versions (Utils 2.4.0, Login 0.11.0, FluxC 1.36.0)

AliSoftware committed:
Audit Release Notes for WordPress vs Jetpack

AliSoftware committed:
Freeze strings for translation

AliSoftware commented:
@leandroalonso Can I ask you for a review of the release notes and validate the audit of the entries I did in aa48e36607f5cbbe373b205773a5e4b860ead8fa ?

In particular this time around:
 - Not only did some entries applied for WP but not JP, but also one entry seem to apply to Jetpack but not WP (first time that happens since the new process I think 😅 )
 - There are entries about Sign Up which I think is not relevant to JP, but there are also entries about site creation, and I'm not fully sure if Jetpack is supposed to allow site creation or not? Similar question for Login epilogue and its behavior in JP

AliSoftware commented on WordPress/jetpack_metadata/release_notes.txt:
```diff
@@ -1,4 +1,7 @@
-- When changing the media in an Image block that’s also a post’s Featured Image, get a prompt to replace the Featured Image.
-- Share text, images, and video in the app using your device’s Share action.
+* [*] Comment editor: Notification comments support in the new editor [https://github.com/wordpress-mobile/WordPress-Android/pull/15884]
+* [*] Site creation: Fixed bug where sites created within the app were not given the correct time zone, leading to post scheduling issues. [https://github.com/wordpress-mobile/WordPress-Android/pull/15904]
```
Doubt: Is site creation applicable to Jetpack? 🤔

ashiagr replied:
Yes, it was decided to enable site creation for Jetpack. Here are corresponding iOS PRs: 
https://github.com/wordpress-mobile/WordPress-iOS/pull/17914
https://github.com/wordpress-mobile/WordPress-iOS/pull/17948

Cc @leandroalonso

AliSoftware replied:
Same question as https://github.com/wordpress-mobile/WordPress-Android/pull/15986#discussion_r811115368 then 😅

ashiagr replied:
Personally, I think we can remove  `Site creation: Fixed bug where sites created...` entry for JP app for the same reasons as https://github.com/wordpress-mobile/WordPress-Android/pull/15986#discussion_r811115368. 😅

AliSoftware replied:
Addressed in [681333f](https://github.com/wordpress-mobile/WordPress-Android/pull/15986/commits/681333ff5bee95e216234470184fd4363154d507)

AliSoftware commented on WordPress/jetpack_metadata/release_notes.txt:
```diff
@@ -1,4 +1,7 @@
-- Share text, images, and video in the app using your device’s Share action.
+* [*] Comment editor: Notification comments support in the new editor [https://github.com/wordpress-mobile/WordPress-Android/pull/15884]
+* [*] Site creation: Fixed bug where sites created within the app were not given the correct time zone, leading to post scheduling issues. [https://github.com/wordpress-mobile/WordPress-Android/pull/15904]
+* [*] Login Epilogue: Fixed bug where if no sites available, then new login epilogue screen without clickable sites and with "create new site" is shown instead of no sites empty "My Site" view. [https://github.com/wordpress-mobile/WordPress-Android/pull/15944]
```
Not entirely sure that the Login Epilogue is the same for WP and JP, and if not how this entry should look like for Jetpack (or if it should be there at all)

ashiagr replied:
As of now, they're the same for both WP, JP as both signup and site creation is enabled for JP now. Similar changes are made on iOS: https://github.com/wordpress-mobile/WordPress-iOS/pull/17914

Cc @leandroalonso

AliSoftware replied:
Same as https://github.com/wordpress-mobile/WordPress-Android/pull/15986#discussion_r811115368 : should we still keep that in the JP release notes anyway even if the feature is now enabled, given that users of Jetpack 19.2 never saw that Login Epilogue before (#17914 not being in 19.2), so telling them that a bug was fixed on something that will be new to them feels strange?

ashiagr replied:
> given that users of Jetpack 19.2 never saw that Login Epilogue before 

Makes sense actually. 😅 We can remove this entry for JP.

AliSoftware replied:
Addressed in [681333f](https://github.com/wordpress-mobile/WordPress-Android/pull/15986/commits/681333ff5bee95e216234470184fd4363154d507)

AliSoftware commented on WordPress/jetpack_metadata/release_notes.txt:
```diff
@@ -1,4 +1,7 @@
+* [*] Fixes flickering when changing the preview mode in Page or Site design picker [https://github.com/wordpress-mobile/WordPress-Android/pull/15943]
+* [**] Jetpack App: Signup using WordPress.com, create your favorite site and enjoy the Reader inside Jetpack app. [https://github.com/wordpress-mobile/WordPress-Android/pull/15946]
+* [*] Block Editor: Stop editor from unexpectedly closing while composing a story block when memory is low [https://github.com/wordpress-mobile/WordPress-Android/pull/15960]
+* [*] Improves the error message shown when trying to create a new site with non-English characters in the domain name [https://github.com/wordpress-mobile/WordPress-Android/pull/15969]
```
Doubt: Is site creation applicable to Jetpack? 🤔

ashiagr replied:
Same as https://github.com/wordpress-mobile/WordPress-Android/pull/15986/commits/aa48e36607f5cbbe373b205773a5e4b860ead8fa#r811087580. Both site creation and signup are enabled for Jetpack.

AliSoftware replied:
Should we pick a different wording for those entries for Jetpack then? Because it seems strange that an existing user of Jetpack, who never saw the flow for creating a new site before since it wasn't enabled in 19.2, would see "improve an error message on… [something that was not an existing feature before]" 🤔  wdyt?

ashiagr replied:
Yeah, I feel the same. Maybe we should remove `Improves the error...`  entry for the JP app.

AliSoftware replied:
Addressed in [681333f](https://github.com/wordpress-mobile/WordPress-Android/pull/15986/commits/681333ff5bee95e216234470184fd4363154d507)

AliSoftware commented on WordPress/jetpack_metadata/release_notes.txt:
```diff
@@ -1,4 +1,7 @@
+* [*] Site creation: Fixed bug where sites created within the app were not given the correct time zone, leading to post scheduling issues. [https://github.com/wordpress-mobile/WordPress-Android/pull/15904]
+* [*] Login Epilogue: Fixed bug where if no sites available, then new login epilogue screen without clickable sites and with "create new site" is shown instead of no sites empty "My Site" view. [https://github.com/wordpress-mobile/WordPress-Android/pull/15944]
+* [*] Fixes flickering when changing the preview mode in Page or Site design picker [https://github.com/wordpress-mobile/WordPress-Android/pull/15943]
+* [**] Jetpack App: Signup using WordPress.com, create your favorite site and enjoy the Reader inside Jetpack app. [https://github.com/wordpress-mobile/WordPress-Android/pull/15946]
```
Note: this one was [removed from the WordPress notes](https://github.com/wordpress-mobile/WordPress-Android/pull/15986/commits/aa48e36607f5cbbe373b205773a5e4b860ead8fa#diff-82636e6ad25e1187592f3ae87c9126c3e71c8a9fdff1a3f23b075456e320915dL6) as it seems to be Jetpack-app-only.

ashiagr replied:
You did it right, thank you! 🙇‍♀️

AliSoftware replied:
Addressed in [681333f](https://github.com/wordpress-mobile/WordPress-Android/pull/15986/commits/681333ff5bee95e216234470184fd4363154d507)

AliSoftware commented on WordPress/jetpack_metadata/release_notes.txt:
```diff
@@ -1,10 +1,7 @@
-* [*] Fixed email app detection in signup for Android 11, and removed the useless button when none are available. [https://github.com/wordpress-mobile/WordPress-Android/pull/15911]
```
@ashiagr does that mean that I was wrong removing that entry then?

ashiagr replied:
I think we can remove this entry from release notes for JP as signup was never enabled until now and JP users need not know if anything was fixed related to signup.

AliSoftware commented on WordPress/jetpack_metadata/release_notes.txt:
```diff
@@ -1,10 +1,7 @@
 * [*] Login Epilogue: Fixed bug where if no sites available, then new login epilogue screen without clickable sites and with "create new site" is shown instead of no sites empty "My Site" view. [https://github.com/wordpress-mobile/WordPress-Android/pull/15944]
 * [*] Fixes flickering when changing the preview mode in Page or Site design picker [https://github.com/wordpress-mobile/WordPress-Android/pull/15943]
 * [**] Jetpack App: Signup using WordPress.com, create your favorite site and enjoy the Reader inside Jetpack app. [https://github.com/wordpress-mobile/WordPress-Android/pull/15946]
-* [*] Reader: Admins and Editors can edit comments from Reader [https://github.com/wordpress-mobile/WordPress-Android/pull/15957]
```
@ashiagr Likewise, should I restore this entry from the JP release notes?

ashiagr replied:
Since `Reader` was never enabled for JP until now, we can remove this entry for JP users as they need not know if anything was fixed for Reader.

AliSoftware replied:
Addressed in d07d573c07. Thanks for the feedback!

AliSoftware reviewed (commented):
Ok so I think I then need to restore some Signup- and Reader-related entries I deleted from the release notes then, now that those are enabled in Jetpack.

@ashiagr can you confirm that the Release Notes for Jetpack should then be ALL of the same items that are in the WordPress ones? (with the only difference being the one about `[Jetpack App] …` that should be only in JP)

[EDIT] Actually I think some items should still be removed from Jetpack Release Notes even if the features are now enabled… because they are about bug fixes on previous iteration of those features, while those features being new to Jetpack, that doesn't make that much sense to me to mention a bugfix on them for Jetpack users.

AliSoftware committed:
Fix release notes for Jetpack

Signup, Site creation and Reader are now available in Jetpack, but they are new in 19.3, so no point in mentioning items that are related to *fixes* on those features since they were not present to Jetpack users in 19.2 before that update.

AliSoftware committed:
Jetpack Release Notes: also remove the entry about new feature in the Reader

Since the Reader was not present in previous versions, mentioning improvements about it makes little sense to Jetpack users

-mAlso splitting the entry about new features in Jetpack in two, to emphasis each (Signup+Site creation, and Reader)

ParaskP7 reviewed (approved):
LGTM! 💯
